Add cuda as a variant to opa-psm2

rfeki opened the pull request:
Enable cuda support on opa-psm2 by setting PSM_CUDA to 1 if the "+cuda" variant is requested.

rfeki committed:
Add cuda as a variant to opa-psm2

Signed-off-by: Raafat Feki <rfeki@cornelisnetworks.com>

jack-morrison reviewed (changes requested):
@rfeki, I think this looks pretty good. 

One small request. It looks like [opa-psm2 with CUDA support should be built using CUDA 8.0 or later](https://github.com/cornelisnetworks/opa-psm2/blob/ad5dd1bed7ba38809ef02e76278ad636194efdfc/README#L109).

If this is correct, do you mind adding a [`depends_on`](https://spack.readthedocs.io/en/latest/packaging_guide.html#depends-on) for this?

Should be something like:
```
depends_on("cuda@8:", when="+cuda")
```

I believe this ought to keep Spack from attempting to build opa-psm2 with an older CUDA.

rfeki committed:
Add Cuda Dependency to opa-psm2 package

gpu support in opa-psm2 requires additional packages for GPU Direct support included in NVIDIA CUDA toolkit 8.0 or greater, hence the dependency to this cuda version.

Signed-off-by: Raafat Feki <rfeki@cornelisnetworks.com>

rfeki commented:
@jack-morrison Done.
